pkg/template/controller/templateinstance_controller.go

```diff
@@ -72,18 +74,19 @@ type TemplateInstanceController struct {
 }
 
 // NewTemplateInstanceController returns a new TemplateInstanceController.
-func NewTemplateInstanceController(config *rest.Config, kc kclientsetinternal.Interface, buildClient buildclient.Interface, templateClient templateclient.Interface, informer internalversion.TemplateInstanceInformer) *TemplateInstanceController {
+func NewTemplateInstanceController(dynamicRestMapper *discovery.DeferredDiscoveryRESTMapper, config *rest.Config, kc kclientsetinternal.Interface, buildClient buildclient.Interface, templateClient templateclient.Interface, informer internalversion.TemplateInstanceInformer) *TemplateInstanceController {
 	c := &TemplateInstanceController{
-		restmapper:       restutil.DefaultMultiRESTMapper(),
-		config:           config,
-		kc:               kc,
-		templateClient:   templateClient,
-		buildClient:      buildClient,
-		lister:           informer.Lister(),
-		informer:         informer.Informer(),
-		queue:            workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "openshift_template_instance_controller"),
-		readinessLimiter: workqueue.NewItemFastSlowRateLimiter(5*time.Second, 20*time.Second, 200),
-		clock:            clock.RealClock{},
+		restmapper:        restutil.DefaultMultiRESTMapper(),
```

Comment:
this restmapper looks compiled in.  Why are you passing one in and using this?

Reply:
may have to talk in person about this one.  I didn't change this code, I only added the dynamicRestMapper.  (and per the discussion in the other PR, we might not need this mapper at all anymore but right now the checkReadiness codepath is using it)